cmd/syft/cli/attest/attest.go

```diff
@@ -1,398 +1,195 @@
+				return
+			}
 
-func generateAttestation(si source.Input, predicate []byte, src *source.Source, sv *sign.SignerVerifier, predicateType string) ([]byte, error) {
-	var h v1.Hash
+			args := []string{"attest", si.UserInput, "--type", "custom", "--predicate", f.Name()}
+			execCmd := exec.Command(cmd, args...)
+			execCmd.Env = os.Environ()
+			execCmd.Env = append(execCmd.Env, "COSIGN_EXPERIMENTAL=1")
 
-	switch si.ImageSource {
-	case image.OciRegistrySource:
-		switch len(src.Image.Metadata.RepoDigests) {
-		case 0:
-			return nil, fmt.Errorf("cannot generate attestation since no repo digests were found; make sure you're passing an OCI registry source for the attest command")
-		case 1:
-			d, err := name.NewDigest(src.Image.Metadata.RepoDigests[0])
+			// bus adapter for ui to hook into stdout via an os pipe
+			r, w, err := os.Pipe()
 			if err != nil {
-				return nil, err
+				errs <- fmt.Errorf("unable to create os pipe: %w", err)
+				return
 			}
+			defer w.Close()
 
-			h, err = v1.NewHash(d.Identifier())
+			b := &busWriter{r: r, w: w}
+			execCmd.Stdout = b
+			execCmd.Stderr = b
+
+			// attest the SBOM
+			err = execCmd.Run()
 			if err != nil {
-				return nil, err
+				errs <- fmt.Errorf("unable to attest SBOM: %w", err)
+				return
 			}
-		default:
-			return nil, fmt.Errorf("cannot generate attestation since multiple repo digests were found for the image: %+v", src.Image.Metadata.RepoDigests)
-		}
-
-	case image.SingularitySource:
-		var err error
-		h, err = v1.NewHash(src.Image.Metadata.ID)
-		if err != nil {
-			return nil, err
 		}
-	}
-
-	sh, err := attestation.GenerateStatement(attestation.GenerateOpts{
-		Predicate: bytes.NewBuffer(predicate),
-		Type:      predicateType,
-		Digest:    h.Hex,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	payload, err := json.Marshal(sh)
-	if err != nil {
-		return nil, err
-	}
 
-	wrapped := dsse.WrapSigner(sv, intotoJSONDsseType)
-	return wrapped.SignMessage(bytes.NewReader(payload), signatureoptions.WithContext(context.Background()))
+		bus.Publish(partybus.Event{
+			Type:  event.Exit,
+			Value: func() error { return nil },
+		})
+	}()
+	return errs
 }
 
-// publishAttestation publishes signedPayload to the location specified by the user.
-func publishAttestation(app *config.Application, signedPayload []byte, src *source.Source, sv *sign.SignerVerifier) error {
-	switch {
-	// We want to give the option to not upload the generated attestation
-	// if passed or if the user is using local PKI
-	case app.Attest.NoUpload || app.Attest.KeyRef != "":
-		if app.File != "" {
-			return os.WriteFile(app.File, signedPayload, 0600)
-		}
-
-		_, err := os.Stdout.Write(signedPayload)
-		return err
-
-	default:
-		ref, err := name.ParseReference(src.Metadata.ImageMetadata.UserInput)
-		if err != nil {
-			return err
-		}
-
-		digest, err := ociremote.ResolveDigest(ref)
-		if err != nil {
-			return err
+func ValidateOutputOptions(app *config.Application) error {
```

Comment:
Todo: I think this function is duplicated - I'll move it so it's in a common spot all cmd can use it